### PR TITLE
feat(codegen,types): SetRelation on expanded types

### PR DESCRIPTION
**Pull request by patrick-medusajs**
## What

On expanded relations, mutate type using SetRelation, an alias for SetRequired with SetNonNullable.

## Why

Simplifies implementation in client code.

## How

* Export SetRelation type util
* Update codegen template

Extra scope:

* Improve codegen error handling when processing x-expanded-relations
* Add `eager` sub-categorization to x-expanded-relations to highlight relations that are loaded using TypeORM eager.

### CODE REVIEW

**Review by riqwan — APPROVED** [review]

File: packages/oas/openapi-typescript-codegen/src/openApi/v3/parser/getModelsExpandedRelations.ts [hunk 1]
```diff
@@ -59,12 +64,20 @@ const walkNestedRelations = (
   nestedRelation: NestedRelation,
   parentNestedRelation?: NestedRelation
 ) => {
-  const prop =
-    model.export === "all-of"
-      ? findPropInAllOf(nestedRelation.field, model, allModels)
-      : getPropertyByName(nestedRelation.field, model)
+  const prop = ["all-of", "any-of", "all-of"].includes(model.export)
+    ? findPropInAllOf(nestedRelation.field, model, allModels)
```
[the remark this answers is not in the document]
Comment (olivermrbl) [reply packages/oas/openapi-typescript-codegen/src/openApi/v3/parser/getModelsExpandedRelations.ts:68]: Also, `all-of` is duplicated

File: packages/oas/openapi-typescript-codegen/src/openApi/v3/parser/getModelsExpandedRelations.ts [hunk 2]
```diff
@@ -59,12 +64,20 @@ const walkNestedRelations = (
...
+    : getPropertyByName(nestedRelation.field, model)
   if (!prop) {
-    return
+    throw new Error(`x-expanded-relations - field not found
+      Schema: ${rootModel.name}
+      NestedRelation: ${JSON.stringify(nestedRelation, null, 2)}
+      Model: ${JSON.stringify(model.spec, null, 2)}`)
+  }
+  if (["all-of", "any-of", "all-of"].includes(prop.export)) {
```
Comment (olivermrbl) [inline packages/oas/openapi-typescript-codegen/src/openApi/v3/parser/getModelsExpandedRelations.ts:76]: todo: `all-of` duplicated

**Review by olivermrbl — APPROVED** [review]

File: packages/oas/openapi-typescript-codegen/src/openApi/v3/parser/getModelsExpandedRelations.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (patrick-medusajs) [reply packages/oas/openapi-typescript-codegen/src/openApi/v3/parser/getModelsExpandedRelations.ts]: Good catch 🎣. Fixed in 1759dbff8d96ac94586c78697804bea6ff865558

File: packages/oas/openapi-typescript-codegen/src/openApi/v3/parser/getModelsExpandedRelations.ts (hunk 2 above)
Comment (patrick-medusajs) [reply packages/oas/openapi-typescript-codegen/src/openApi/v3/parser/getModelsExpandedRelations.ts]: Good catch 🎣. Fixed in 1759dbff8d96ac94586c78697804bea6ff865558

**Pull request merged**